src/Microsoft.Diagnostics.Monitoring.Options/MonitorCapability.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Diagnostics.Monitoring.Options
+{
+    public class MonitorCapability : IMonitorCapability
```

Comment:
I'm not sure if this class adds value here. We already have `IMonitorCapability` and we can have simple concrete implementations for each capability. Each concrete implementation would then be able to have isolated logic to determine if the feature is on or not (some abstraction may be wanted for in-proc features).

Reply:
@jander-msft can weigh in, but if I'm understanding this correctly that's more aligned with the initial approach I took, and then we transitioned to this model that made everything standardized

Reply:
Maybe a compromise here would be that the class structure stays the same but the default implementing class doesn't have upfront knowledge about all of its capabilities. This could be done at the registration level.

Reply:
We can also make the interface public but the class internal